### PR TITLE
Scheduler JobQueue debugger

### PR DESCRIPTION
**Pull request by anwar6953**
Reopening https://github.com/caskdata/cdap/pull/8772.

https://builds.cask.co/browse/CDAP-RUT1062-1

Example output:
```
$ sudo -u cdap /opt/cdap/master/bin/cdap run co.cask.cdap.data.tools.JobQueueDebugger
[WARN] Fail to source spark-env.sh to setup environment variables for Spark
Tue May 16 20:29:39 UTC 2017 Running class co.cask.cdap.data.tools.JobQueueDebugger
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/opt/cdap/master/lib/ch.qos.logback.logback-classic-1.0.9.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/usr/hdp/2.3.4.7-4/hadoop/lib/slf4j-log4j12-1.7.10.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/usr/hdp/2.3.4.7-4/zookeeper/lib/slf4j-log4j12-1.6.1.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [ch.qos.logback.classic.util.ContextSelectorStaticBinder]
2017-05-16 20:29:42,049 - WARN  [main:c.c.c.c.c.Configuration@624] - log.retention.duration.days is deprecated. Instead, use log.pipeline.cdap.file.retention.duration.days
2017-05-16 20:29:42,054 - WARN  [main:c.c.c.c.c.Configuration@624] - ssl.enabled is deprecated. Instead, use ssl.external.enabled
2017-05-16 20:29:42,055 - WARN  [main:c.c.c.c.c.Configuration@624] - security.auth.server.address is deprecated. Instead, use security.auth.server.bind.address
2017-05-16 20:29:42,429 - WARN  [main:o.a.h.u.NativeCodeLoader@62] - Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
2017-05-16 20:29:42,612 - INFO  [main:c.c.c.s.i.SecurityUtil@166] - Logging in as: principal=cdap/<HOSTNAME>@<REALM>, keytab=/etc/security/keytabs/cdap.service.keytab
2017-05-16 20:29:44,587 - INFO  [authorization-enforcement-service:c.c.c.s.a.AbstractAuthorizationService@102] - Started authorization enforcement service...
Getting notification subscriber messageIds.
Topic: timeevent, Publish Timestamp: 1494963180010
Topic: dataevent, Publish Timestamp: 1494966399451

Scanning JobQueue.
Scanning partition id 0.
N/A - empty partition
Scanning partition id 1.
N/A - empty partition
Scanning partition id 2.
N/A - empty partition
Scanning partition id 3.
N/A - empty partition
Scanning partition id 4.
N/A - empty partition
Scanning partition id 5.
N/A - empty partition
Scanning partition id 6.
N/A - empty partition
Scanning partition id 7.
N/A - empty partition
Scanning partition id 8.
Number of Jobs by state:
  Pending Trigger: 2
  Pending Constraint: 0
  Pending Launch: 0
  Total: 2

Scanning partition id 9.
N/A - empty partition
Scanning partition id 10.
N/A - empty partition
Scanning partition id 11.
N/A - empty partition
Scanning partition id 12.
N/A - empty partition
Scanning partition id 13.
N/A - empty partition
Scanning partition id 14.
N/A - empty partition
Scanning partition id 15.
N/A - empty partition

Total statistics:
Number of Jobs by state:
  Pending Trigger: 2
  Pending Constraint: 0
  Pending Launch: 0
  Total: 2

2017-05-16 20:29:46,966 - INFO  [authorization-enforcement-service:c.c.c.s.a.AbstractAuthorizationService@139] - Shutdown authorization enforcement service successfully.
```

Example trace output:
```
$ sudo -u cdap /opt/cdap/master/bin/cdap run co.cask.cdap.data.tools.JobQueueDebugger -t
[ERROR] Failed to get Spark and Scala versions using spark-shell
  stderr:
/opt/cdap/master/bin/functions.sh: line 575: /bin/spark-shell: No such file or directory
[WARN] Fail to source spark-env.sh to setup environment variables for Spark
Wed May 31 18:01:38 UTC 2017 Running class co.cask.cdap.data.tools.JobQueueDebugger with arguments: -t
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/opt/cdap/master/lib/ch.qos.logback.logback-classic-1.0.9.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/usr/hdp/2.3.4.7-4/hadoop/lib/slf4j-log4j12-1.7.10.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/usr/hdp/2.3.4.7-4/zookeeper/lib/slf4j-log4j12-1.6.1.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [ch.qos.logback.classic.util.ContextSelectorStaticBinder]
2017-05-31 18:01:41,106 - WARN  [main:c.c.c.c.c.Configuration@624] - log.retention.duration.days is deprecated. Instead, use log.pipeline.cdap.file.retention.duration.days
2017-05-31 18:01:41,112 - WARN  [main:c.c.c.c.c.Configuration@624] - ssl.enabled is deprecated. Instead, use ssl.external.enabled
2017-05-31 18:01:41,112 - WARN  [main:c.c.c.c.c.Configuration@624] - security.auth.server.address is deprecated. Instead, use security.auth.server.bind.address
2017-05-31 18:01:41,571 - WARN  [main:o.a.h.u.NativeCodeLoader@62] - Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
2017-05-31 18:01:41,814 - INFO  [main:c.c.c.s.i.SecurityUtil@166] - Logging in as: principal=cdap/<HOSTNAME>@<REALM>, keytab=/etc/security/keytabs/cdap.service.keytab
Getting notification subscriber messageIds.
Topic: timeevent, Publish Timestamp: 1496253660002
Topic: dataevent, Publish Timestamp: 1496253703824

Scanning JobQueue.
Scanning partition id 0.
N/A - empty partition
Scanning partition id 1.
N/A - empty partition
Scanning partition id 2.
N/A - empty partition
Scanning partition id 3.
N/A - empty partition
Scanning partition id 4.
N/A - empty partition
Scanning partition id 5.
Job: {"schedule":{"description":"","programId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","type":"Workflow","program":"TimeTriggerOnlyWorkflow","namespace":"ali","entity":"PROGRAM"},"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"AlwaysFailSchedule","namespace":"ali","entity":"SCHEDULE"},"properties":{"fail_key":"fail_value"},"trigger":{"dataset":{"dataset":"rawRecords","namespace":"ali","entity":"DATASET"},"numPartitions":5,"type":"PARTITION"},"constraints":[]},"jobKey":{"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"AlwaysFailSchedule","namespace":"ali","entity":"SCHEDULE"},"creationTime":1496253701878},"notifications":[{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253700679]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253700814]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253700919]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253701436]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253701567]}]","datasetId":"dataset:ali.rawRecords"}}],"state":"PENDING_LAUNCH","scheduleLastUpdatedTime":1496253568620}
Job: {"schedule":{"description":"","programId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","type":"Workflow","program":"TimeTriggerOnlyWorkflow","namespace":"ali","entity":"PROGRAM"},"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"AlwaysFailSchedule","namespace":"ali","entity":"SCHEDULE"},"properties":{"fail_key":"fail_value"},"trigger":{"dataset":{"dataset":"rawRecords","namespace":"ali","entity":"DATASET"},"numPartitions":5,"type":"PARTITION"},"constraints":[]},"jobKey":{"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"AlwaysFailSchedule","namespace":"ali","entity":"SCHEDULE"},"creationTime":1496253703993},"notifications":[{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253702190]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253702339]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253702636]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253703388]}]","datasetId":"dataset:ali.rawRecords"}}],"state":"PENDING_TRIGGER","scheduleLastUpdatedTime":1496253568620}
Job: {"schedule":{"description":"","programId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","type":"Workflow","program":"TwoActionsWorkflow","namespace":"ali","entity":"PROGRAM"},"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"ConCurrencySchedule","namespace":"ali","entity":"SCHEDULE"},"properties":{},"trigger":{"dataset":{"dataset":"rawRecords","namespace":"ali","entity":"DATASET"},"numPartitions":5,"type":"PARTITION"},"constraints":[{"maxConcurrency":1,"type":"CONCURRENCY","waitUntilMet":false}]},"jobKey":{"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"ConCurrencySchedule","namespace":"ali","entity":"SCHEDULE"},"creationTime":1496253704002},"notifications":[{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253702190]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253702339]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253702636]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253703388]}]","datasetId":"dataset:ali.rawRecords"}}],"state":"PENDING_TRIGGER","scheduleLastUpdatedTime":1496253568359}
Job: {"schedule":{"description":"","programId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","type":"Workflow","program":"TimeTriggerOnlyWorkflow","namespace":"mao","entity":"PROGRAM"},"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"OneMinSchedule","namespace":"mao","entity":"SCHEDULE"},"properties":{},"trigger":{"cronExpression":"* * * * *","type":"TIME"},"constraints":[{"millisSinceLastRun":600000,"type":"LAST_RUN","waitUntilMet":false}]},"jobKey":{"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"OneMinSchedule","namespace":"mao","entity":"SCHEDULE"},"creationTime":1496253723702},"notifications":[{"notificationType":"TIME","properties":{"userOverrides":"{\"logical.start.time\":\"1496253720000\"}","scheduleId":"schedule:mao.AppWithDataPartitionSchedule.-SNAPSHOT.OneMinSchedule","systemOverrides":"{\"scheduleName\":\"OneMinSchedule\"}"}}],"state":"PENDING_CONSTRAINT","scheduleLastUpdatedTime":1496217571577}
Number of Jobs by state:
  Pending Trigger: 2
  Pending Constraint: 1
  Pending Launch: 1
  Total: 4

Scanning partition id 6.
N/A - empty partition
Scanning partition id 7.
N/A - empty partition
Scanning partition id 8.
Job: {"schedule":{"description":"","programId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","type":"Workflow","program":"TimeTriggerOnlyWorkflow","namespace":"ali","entity":"PROGRAM"},"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"OneMinSchedule","namespace":"ali","entity":"SCHEDULE"},"properties":{},"trigger":{"cronExpression":"* * * * *","type":"TIME"},"constraints":[{"millisSinceLastRun":600000,"type":"LAST_RUN","waitUntilMet":false}]},"jobKey":{"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"OneMinSchedule","namespace":"ali","entity":"SCHEDULE"},"creationTime":1496253723691},"notifications":[{"notificationType":"TIME","properties":{"userOverrides":"{\"logical.start.time\":\"1496253720000\"}","scheduleId":"schedule:ali.AppWithDataPartitionSchedule.-SNAPSHOT.OneMinSchedule","systemOverrides":"{\"scheduleName\":\"OneMinSchedule\"}"}}],"state":"PENDING_CONSTRAINT","scheduleLastUpdatedTime":1496253568782}
Number of Jobs by state:
  Pending Trigger: 0
  Pending Constraint: 1
  Pending Launch: 0
  Total: 1

Scanning partition id 9.
N/A - empty partition
Scanning partition id 10.
N/A - empty partition
Scanning partition id 11.
Job: {"schedule":{"description":"","programId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","type":"Workflow","program":"DelayWorkflow","namespace":"ali","entity":"PROGRAM"},"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"DelaySchedule","namespace":"ali","entity":"SCHEDULE"},"properties":{},"trigger":{"dataset":{"dataset":"rawRecords","namespace":"ali","entity":"DATASET"},"numPartitions":5,"type":"PARTITION"},"constraints":[{"millisAfterTrigger":5000,"type":"DELAY","waitUntilMet":true}]},"jobKey":{"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"DelaySchedule","namespace":"ali","entity":"SCHEDULE"},"creationTime":1496253697725},"notifications":[{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253696222]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253696335]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253698273]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253698383]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253698491]}]","datasetId":"dataset:ali.rawRecords"}}],"state":"PENDING_LAUNCH","scheduleLastUpdatedTime":1496253568492}
Job: {"schedule":{"description":"","programId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","type":"Workflow","program":"DelayWorkflow","namespace":"ali","entity":"PROGRAM"},"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"DelaySchedule","namespace":"ali","entity":"SCHEDULE"},"properties":{},"trigger":{"dataset":{"dataset":"rawRecords","namespace":"ali","entity":"DATASET"},"numPartitions":5,"type":"PARTITION"},"constraints":[{"millisAfterTrigger":5000,"type":"DELAY","waitUntilMet":true}]},"jobKey":{"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"DelaySchedule","namespace":"ali","entity":"SCHEDULE"},"creationTime":1496253699802},"notifications":[{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253699447]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253699553]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253699699]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253699842]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253699993]}]","datasetId":"dataset:ali.rawRecords"}}],"state":"PENDING_CONSTRAINT","scheduleLastUpdatedTime":1496253568492}
Job: {"schedule":{"description":"","programId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","type":"Workflow","program":"DelayWorkflow","namespace":"ali","entity":"PROGRAM"},"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"DelaySchedule","namespace":"ali","entity":"SCHEDULE"},"properties":{},"trigger":{"dataset":{"dataset":"rawRecords","namespace":"ali","entity":"DATASET"},"numPartitions":5,"type":"PARTITION"},"constraints":[{"millisAfterTrigger":5000,"type":"DELAY","waitUntilMet":true}]},"jobKey":{"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"DelaySchedule","namespace":"ali","entity":"SCHEDULE"},"creationTime":1496253701880},"notifications":[{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253700679]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253700814]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253700919]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253701436]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253701567]}]","datasetId":"dataset:ali.rawRecords"}}],"state":"PENDING_CONSTRAINT","scheduleLastUpdatedTime":1496253568492}
Job: {"schedule":{"description":"","programId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","type":"Workflow","program":"DelayWorkflow","namespace":"ali","entity":"PROGRAM"},"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"DelaySchedule","namespace":"ali","entity":"SCHEDULE"},"properties":{},"trigger":{"dataset":{"dataset":"rawRecords","namespace":"ali","entity":"DATASET"},"numPartitions":5,"type":"PARTITION"},"constraints":[{"millisAfterTrigger":5000,"type":"DELAY","waitUntilMet":true}]},"jobKey":{"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"DelaySchedule","namespace":"ali","entity":"SCHEDULE"},"creationTime":1496253704001},"notifications":[{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253702190]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253702339]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253702636]}]","datasetId":"dataset:ali.rawRecords"}},{"notificationType":"PARTITION","properties":{"numPartitions":"1","partitionKeys":"[{\"time\":[\"java.lang.Long\",1496253703388]}]","datasetId":"dataset:ali.rawRecords"}}],"state":"PENDING_TRIGGER","scheduleLastUpdatedTime":1496253568492}
Number of Jobs by state:
  Pending Trigger: 1
  Pending Constraint: 2
  Pending Launch: 1
  Total: 4

Scanning partition id 12.
Job: {"schedule":{"description":"","programId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","type":"Workflow","program":"TimeTriggerOnlyWorkflow","namespace":"default","entity":"PROGRAM"},"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"OneMinSchedule","namespace":"default","entity":"SCHEDULE"},"properties":{},"trigger":{"cronExpression":"* * * * *","type":"TIME"},"constraints":[{"millisSinceLastRun":600000,"type":"LAST_RUN","waitUntilMet":false}]},"jobKey":{"scheduleId":{"application":"AppWithDataPartitionSchedule","version":"-SNAPSHOT","schedule":"OneMinSchedule","namespace":"default","entity":"SCHEDULE"},"creationTime":1496253723699},"notifications":[{"notificationType":"TIME","properties":{"userOverrides":"{\"logical.start.time\":\"1496253720000\"}","scheduleId":"schedule:default.AppWithDataPartitionSchedule.-SNAPSHOT.OneMinSchedule","systemOverrides":"{\"scheduleName\":\"OneMinSchedule\"}"}}],"state":"PENDING_CONSTRAINT","scheduleLastUpdatedTime":1496220632602}
Number of Jobs by state:
  Pending Trigger: 0
  Pending Constraint: 1
  Pending Launch: 0
  Total: 1

Scanning partition id 13.
N/A - empty partition
Scanning partition id 14.
N/A - empty partition
Scanning partition id 15.
N/A - empty partition

Total statistics:
Number of Jobs by state:
  Pending Trigger: 3
  Pending Constraint: 5
  Pending Launch: 2
  Total: 10

```

### CODE REVIEW

File: cdap-master/src/main/java/co/cask/cdap/data/tools/JobQueueDebugger.java [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+      new KafkaClientModule(),
+      new NamespaceStoreModule().getDistributedModules(),
+      new AuthorizationModule(),
+      new AuthorizationEnforcementModule().getMasterModule(),
+      new SecureStoreModules().getDistributedModules(),
+      new MessagingClientModule(),
+      new AbstractModule() {
+        @Override
+        protected void configure() {
+          bind(QueueClientFactory.class).to(HBaseQueueClientFactory.class).in(Singleton.class);
```
Comment (anew) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/JobQueueDebugger.java]: do we need these?

**Review by anew — COMMENTED** [review]
only one minor comment.

**Comment by anew** [comment]
Can you paste a sample output for -t (with trace)?

File: cdap-master/src/main/java/co/cask/cdap/data/tools/JobQueueDebugger.java (hunk 1 above)
Comment (anwar6953) [reply cdap-master/src/main/java/co/cask/cdap/data/tools/JobQueueDebugger.java]: no; removed.

**Comment by anwar6953** [comment]
Attached example trace output in the PR description.

**Review by anew — COMMENTED** [review]
LGTM

**Pull request merged**